File: Backend/Scheduler/Model/OrderReport/DetailsBatchBlock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Scheduler.Model.OrderReport
{
    public class DetailsBatchBlock
    {
        public int Id { get; set; }

        public int DetailId { get; set; }

        public string DetailName { get; set; }

        public int DetailsCount { get; set; }

        public int GroupBlockId { get; set; }
        public virtual GroupBlock GroupBlock { get; set; }

        public long StartTime { get; set; }

        public long Duration { get; set; }

        public virtual Equipment Equipment { get; set; }
        public int EquipmentId { get; set; }
    }
}

File: Frontend/HttpClient/src/app/models/Reporting/DetailsBatchBlock.ts
import { Equipment } from '../equipment';

export class DetailsBatchBlock {
    public id: number;

    public detailId: number;

    public detailName: string;

    public detailsCount: number;

    public startTime: number;

    public duration: number;

    public equipment: Equipment;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }

        return Object.assign(
            new DetailsBatchBlock(),
            obj,
            {
                equipment: obj.equipment ? Equipment.fromJSON(obj.equipment) : null
            }
        );
    }
}
